operator/pkg/ciliumendpointbatch/manager.go

```diff
@@ -249,55 +282,46 @@ func (c *cebManagerFcfs) getCebCopyFromCache(cebName string) (*cilium_v2.CiliumE
 	return nil, fmt.Errorf("Failed to get CEB Copy from local cache for the cebName: %s", cebName)
 }
 
-// findCeb returns the available cebTracker object and is chosen based on FCFS.
-// if all CEB's reached max capacity or marked for delete. Allocate a new CEB.
-func (c *cebManagerFcfs) findCeb() (ceb *cebTracker) {
-
-	// get first available CEB
-	for _, ceb = range c.desiredCebs {
-		ceb.backendMutex.RLock()
-		if len(ceb.ceb.Endpoints) >= c.maxCepsInCeb || len(ceb.ceb.Endpoints) == 0 {
-			ceb.backendMutex.RUnlock()
-			continue
-		}
-		ceb.backendMutex.RUnlock()
-		return
-	}
-
-	// allocate a new cebTracker and return
-	ceb = c.createCeb("")
-	return
-}
-
-// InsertCepInCache is used to Insert CEP in local cache, this may result in Creating a New
-// CEB object or Updating an existing CEB object.
-func (c *cebManagerFcfs) InsertCepInCache(cep *cilium_v2.CoreCiliumEndpoint) string {
+// InsertCepInCache is used to insert CEP in local cache, this may result in creating a new
+// CEB object or updating an existing CEB object.
+func (c *cebMgr) InsertCepInCache(cep *cilium_v2.CoreCiliumEndpoint) string {
 
 	// check the given cep is already exists in any of the CEB.
 	// if yes, Update a ceb with the given cep object.
 	if cebName, exists := c.cacheCepMap.get(GetCepNameFromCCEP(cep)); exists {
 		// add a cep into the ceb
-		addCEPtoCEB(cep, c.desiredCebs[cebName])
-		c.queue.Add(cebName)
+		c.addCEPtoCEB(cep, cebName)
 		return cebName
 	}
 
 	// If given cep object isn't packed in any of the CEB. find a new ceb
 	// to pack this cep.
-	cb := c.findCeb()
+	cb := func() *cebTracker {
```

Comment:
why does this need to be in a function?

Reply:
Preferred keeping `InsertCepInCache` function simple.

Reply:
then maybe put this common code into its own function? I don't see how nesting the code into a local function which in only called here once makes `InsertCepInCache` simpler...

Reply:
I don't think this fits for a separate function, because that function only used by `InsertCepInCache`.  Having `anonymous` function fit well here. If you look Identity based `InsertCepInCache`, it is quite different from this one.